### PR TITLE
DataViews: Update `usePostFields` to accept postType

### PR DESCRIPTION
**Pull request by ntsekouras**
## What?
Part of: https://github.com/WordPress/gutenberg/issues/61084

This PR updates `usePostFields` to accept postType and add some support checks to conditionally add some fields, like featured image.

This surfaces some issues we need to solve regarding layouts using some fields that might not exist. For example to handle when we don't support `featured image` but `view.layout.mediaField` is declared to show that. These should be handled in follow ups to iterate faster.

## Testing Instructions
1. Everything should work as before in trunk
2. Test `wp.editor.unregisterEntityField( 'postType', 'page', 'author' );` or `wp.editor.unregisterEntityField( 'postType', 'post', 'author' );` in the respective DataViews lists

### CODE REVIEW

File: packages/editor/src/dataviews/store/private-actions.ts [hunk 1]
```diff
@@ -157,25 +157,24 @@ export const registerPostTypeSchema =
...
 			permanentlyDeletePost,
-		];
+		].filter( Boolean );
 
 		const fields = [
-			featuredImageField,
+			postTypeConfig.supports?.thumbnail &&
+				currentTheme?.[ 'theme-supports' ]?.[ 'post-thumbnails' ] &&
+				featuredImageField,
 			titleField,
```
Comment (ntsekouras) [inline packages/editor/src/dataviews/store/private-actions.ts:166]: I intentionally didn't add the post type supports for title here, as I need to investigate better how core lists (wp-admin) handles it and what we should do here.

Additionally it relates to the issue about layout fields I mentioned in the PR's description. Some layouts declare this as primary field.

File: packages/edit-site/src/components/post-edit/index.js [hunk 2]
```diff
@@ -49,7 +49,8 @@ function PostEditForm( { postType, postId } ) {
 	);
 	const [ multiEdits, setMultiEdits ] = useState( {} );
 	const { editEntityRecord } = useDispatch( coreDataStore );
-	const { fields: _fields } = usePostFields();
+	// TODO: This needs to be dynamic and will be handled in a follow up with the `format` field for posts.
+	const { fields: _fields } = usePostFields( { postType: 'page' } );
```
Comment (youknowriad) [inline packages/edit-site/src/components/post-edit/index.js]: Why not pass postType here? I don't understand

File: packages/editor/src/dataviews/store/private-actions.ts [hunk 3]
```diff
@@ -157,25 +157,24 @@ export const registerPostTypeSchema =
...
 			authorField,
 			statusField,
 			dateField,
 			slugField,
```
Comment (youknowriad) [inline packages/editor/src/dataviews/store/private-actions.ts:170]: This one has a support key too.

File: packages/editor/src/dataviews/store/private-actions.ts [hunk 4]
```diff
@@ -157,25 +157,24 @@ export const registerPostTypeSchema =
...
-		];
+		].filter( Boolean );
 
 		const fields = [
-			featuredImageField,
+			postTypeConfig.supports?.thumbnail &&
+				currentTheme?.[ 'theme-supports' ]?.[ 'post-thumbnails' ] &&
+				featuredImageField,
 			titleField,
 			authorField,
```
Comment (youknowriad) [inline packages/editor/src/dataviews/store/private-actions.ts]: This one has a support key as well.

File: packages/edit-site/src/components/post-edit/index.js (hunk 2 above)
Comment (ntsekouras) [reply packages/edit-site/src/components/post-edit/index.js]: The main reason for now, is that if I remember correctly we only had the `quick edit` only for `pages`. I don't remember why though and that was the reason for now. I need to check this better..

Comment (youknowriad) [reply packages/edit-site/src/components/post-edit/index.js]: Sure, but we should still pass the postType argument here. Whether we have quick edit for pages or posts is controlled elsewhere. (Whether PostEdit is rendered)

File: packages/editor/src/dataviews/store/private-actions.ts (hunk 4 above)
Comment (ntsekouras) [reply packages/editor/src/dataviews/store/private-actions.ts]: I missed that. Will add it.

File: packages/editor/src/dataviews/store/private-actions.ts (hunk 3 above)
Comment (ntsekouras) [reply packages/editor/src/dataviews/store/private-actions.ts:170]: The `slug`?

Comment (youknowriad) [reply packages/editor/src/dataviews/store/private-actions.ts:170]: yes, you can check the different `<PostTypeSupportCheck` calls we have to find these.

Comment (ntsekouras) [reply packages/editor/src/dataviews/store/private-actions.ts:170]: Hm.. the [docs for supports ](https://developer.wordpress.org/reference/functions/post_type_supports/)don't say that.. Are you sure there is such support?

I saw we have a `<PostTypeSupportCheck supportKeys="slug">` check, but I don't think it ever worked properly - unless there was such support in the past. Also I think the above code might not being used anywhere by quickly adding a breakpoint there. I'll check more to see if we can remove that code.

Comment (youknowriad) [reply packages/editor/src/dataviews/store/private-actions.ts:170]: I think you're right :)

**Review by youknowriad — APPROVED** [review]

File: packages/editor/src/dataviews/store/private-actions.ts (hunk 1 above)
Comment (oandregal) [reply packages/editor/src/dataviews/store/private-actions.ts:166]: ~Can you add a TODO in the code so we don't forget?~ Let's ship this.

**Comment by oandregal** [comment]
I see there's a `custom-fields` support, and I wonder how that should be connected to the API we have to register new fields: `wp.editor.registerEntityField`.

**Review by oandregal — APPROVED** [review]
Thanks! :)

I'm seeing some weirds things (quickedit not opening, click behavior for table) but they also happen in `trunk`. See [report in core-editor slack channel](https://wordpress.slack.com/archives/C02QB2JS7/p1732799172907269).

**Pull request merged**